SqlInstall.ps1

```diff
@@ -0,0 +1,433 @@
+        Write-Message -Level Verbose -Message 'Setup will start from ' + $SetupFile
+    }
+
+    # Check if there are designated drives for Data, Log, TempDB, Back-up and Application.
+    If ($DataVolume -eq $null -or $DataVolume -eq '') {
+        $DataVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*Data*'} | 
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($LogVolume -eq $null -or $LogVolume -eq '') {
+        $LogVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*Log*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($TempVolume -eq $null -or $TempVolume -eq '') {
+        $TempVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*TempDB*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($AppVolume -eq $null -or $AppVolume -eq '') {
+        $AppVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*App*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($BackupVolume -eq $null -or $BackupVolume -eq '') {
+        $BackupVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*Backup*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    #Check the number of cores available on the server. Summed because every processor can contain multiple cores
+    $NumberOfCores = Get-CimInstance -ClassName Win32_processor | Measure-Object NumberOfCores -Sum | Select-Object -ExpandProperty sum
+
+    IF ($NumberOfCores -gt 8)
+    { $NumberOfCores = 8 }
+
+    IF ($null -eq $DataVolume -or $DataVolume -eq '') {
+        $DataVolume = 'C'
+    }
+
+    IF ($null -eq $LogVolume -or $LogVolume -eq '') {
+        $LogVolume = $DataVolume
+    }
+
+    IF ( $null -eq $TempVolume -or $TempVolume -eq '') {
+        $TempVolume = $DataVolume
+    }
+
+    IF ( $null -eq $AppVolume -or $AppVolume -eq '') {
+        $AppVolume = 'C'
+    }
+
+    IF ( $null -eq $BackupVolume -or $BackupVolume -eq '') {
+        $BackupVolume = $DataVolume
+    }
+
+
+    Write-Message -Level Verbose -Message 'Your datadrive:' $DataVolume
+    Write-Message -Level Verbose -Message 'Your logdrive:' $LogVolume
+    Write-Message -Level Verbose -Message 'Your TempDB drive:' $TempVolume
+    Write-Message -Level Verbose -Message 'Your applicationdrive:' $AppVolume
+    Write-Message -Level Verbose -Message 'Your Backup Drive:' $BackupVolume
+    Write-Message -Level Verbose -Message 'Number of cores for your Database:' $NumberOfCores
+
+    Write-Message -Level Verbose -Message  'Do you agree on the drives?'
+    $AlterDir = Read-Host " ( Y / N )"
```

Comment:
This will be taken care of by something kind of advanced called `SupportsShouldProcess`. Please remove it for now and we'll revisit once you are closer to your goal. It took me a while to understand `SupportsShouldProcess` and I want to make sure I have the time to make it approachable.